packages/docusaurus-plugin-content-docs/src/routes.ts

```diff
@@ -226,6 +226,14 @@ export async function buildAllRoutes(
       }),
     ),
   );
+  // Add a catch-all route for 404 support if routeBasePath is "/".
+  // (see https://github.com/facebook/docusaurus/issues/9688)
+  if (param.options.routeBasePath === '/') {
+    subRoutes.push({
+      path: '/*',
+      component: '@docusaurus/ComponentCreator',
```

Comment:
What's the benefit of using `ComponentCreator` here instead of `NotFound/Content`? Theoretically, couldn't the `ComponentCreator("*")` at the end of the routes just use `@theme/NotFound`?